Content.Shared/Burial/BurialSystem.cs

```diff
@@ -0,0 +1,176 @@
+    {
+        RemComp<ActiveGraveComponent>(uid);
+
+        if (args.Cancelled || args.Handled)
+            return;
+
+        component.DiggingComplete = true;
+
+        if (args.Used != null)
+            _storageSystem.ToggleOpen(args.User, uid);
+        else
+            _storageSystem.TryOpenStorage(args.User, uid); //can only claw out
+    }
+
+    private void StartDigging(EntityUid uid, EntityUid user, EntityUid? used, GraveComponent component)
+    {
+        ActiveGraveComponent activeComp = EnsureComp<ActiveGraveComponent>(uid);
+
+        if (_gameTiming.IsFirstTimePredicted)
+        {
+            if (used != null)
+            {
+                _popupSystem.PopupEntity(Loc.GetString("grave-start-digging-others", ("user", user), ("grave", uid), ("tool", used)), user, Filter.PvsExcept(user), true);
+                if (_gameTiming.InPrediction)
+                {
+                    _popupSystem.PopupEntity(Loc.GetString("grave-start-digging-user", ("grave", uid), ("tool", used)), user, user);
+                    activeComp.Stream = _audioSystem.PlayPvs(component.DigSound, uid).Value.Entity;
+                }
+            }
+            else
+            {
+                activeComp.DiggingSelfOut = true;
+                _popupSystem.PopupClient(Loc.GetString("grave-start-digging-user-trapped", ("grave", uid)), user, user, PopupType.Medium);
+            }
+        }
+    }
+
+    private void OnStopDigging(EntityUid uid, ActiveGraveComponent component, ComponentShutdown args)
+    {
+        component.Stream = _audioSystem.Stop(component.Stream);
+    }
+
+    private void OnOpenAttempt(EntityUid uid, GraveComponent component, ref StorageOpenAttemptEvent args)
+    {
+        if (component.DiggingComplete)
+            return;
+
+        args.Cancelled = true;
+    }
+
+    private void OnCloseAttempt(EntityUid uid, GraveComponent component, ref StorageCloseAttemptEvent args)
+    {
+        if (component.DiggingComplete)
+            return;
+
+        args.Cancelled = true;
+    }
+
+    private void OnAfterOpen(EntityUid uid, GraveComponent component, ref StorageAfterOpenEvent args)
+    {
+        component.DiggingComplete = false;
+    }
+
+    private void OnAfterClose(EntityUid uid, GraveComponent component, ref StorageAfterCloseEvent args)
+    {
+        component.DiggingComplete = false;
+    }
+
+    private void OnRelayMovement(EntityUid uid, GraveComponent component, ref ContainerRelayMovementEntityEvent args)
+    {
+        if (!HasComp<ActiveGraveComponent>(uid))
+        {
+            _doAfterSystem.TryStartDoAfter(new DoAfterArgs(EntityManager, args.Entity, component.DigDelay / component.DigOutByHandModifier, new GraveDiggingDoAfterEvent(), uid, target: uid)
+            {
+                NeedHand = false,
+                BreakOnUserMove = true,
+                BreakOnTargetMove = false,
+                BreakOnHandChange = false,
+                RequireCanInteract = false
```

Comment:
Why doesn't this require CanInteract? People without hands or any other mod would still be able to commence do_afters.